src/main/bootstrap/mod.rs

```diff
@@ -160,14 +161,7 @@ impl<UID: Uid> Bootstrap<UID> {
         let _ = self.children.remove(&child);
         match res {
             Ok((socket, peer_info, peer_id)) => {
-                {
-                    let bootstrap_cache = core.user_data_mut();
-                    bootstrap_cache.put(peer_info);
-                    if let Err(e) = bootstrap_cache.commit() {
-                        info!("Failed to write bootstrap cache to disk: {}", e);
-                    }
-                }
-
+                cache_peer_info(core, peer_info, &self.config);
```

Comment:
Why do we want to cache folks here ? If they are in HCC, we don't want to cache them. If they are due to ServDisc, we don't want to cache them (likely LAN IP anyway and temporary too, plus ServDisc by itself is fine in (re)finding them). If they are from bootstrap cache, well then they are already in the cache.

Rather what we should be doing is to remove the ones who fail to be contacted which i can see is already implemented in the failure arm, so that should be sufficient ?